### PR TITLE
Added fix for https://github.com/oxidecomputer/typify/issues/125

### PR DESCRIPTION
**Pull request by justinmmott**
Further added partialeq trait derive for new types with enums because of the comparison that is done

### CODE REVIEW

File: typify-impl/src/value.rs [hunk 1]
```diff
@@ -60,7 +60,7 @@ impl TypeEntry {
                     .unwrap()
                     .output_value(type_space, value);
                 let ident = format_ident!("{}", name);
-                quote! { super::#ident ( #inner )}
+                quote! { #ident ( #inner )}
```
Comment (ahl) [inline typify-impl/src/value.rs:63]: related to #137, I think that the "path" needs to be a parameter to `output_value` -- simply deleting `super::` will break other cases (that apparently are untested!)

File: typify-impl/tests/new_type_with_enum.json [hunk 2]
```diff
@@ -0,0 +1,14 @@
+{
+  "$id": "https://example.com/arrays.schema.json",
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "title": "Everything",
+  "type": "object",
+  "properties": {
+    "enum_property": {
+      "description": "name",
+      "type": "string",
+      "maxLength": 32,
```
Comment (ahl) [inline typify-impl/tests/new_type_with_enum.json:10]: This is a case where I think we should simply ignore `maxLength` or use it to cull the list of value `enum` values.

File: typify-impl/tests/new_type_with_enum.out [hunk 3]
```diff
@@ -0,0 +1,69 @@
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct Everything {
+    #[doc = "name"]
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub enum_property: Option<EverythingEnumProperty>,
+}
+#[doc = "name"]
+#[derive(Clone, Debug, Deserialize, PartialEq, Serialize)]
+pub struct EverythingEnumProperty(EverythingEnumPropertyInner);
```
Comment (ahl) [inline typify-impl/tests/new_type_with_enum.out:9]: I really want `EverythingEnumProperty` in this case to be an `enum`

**Review by ahl — CHANGES_REQUESTED** [review]

**Comment by jayvdb** [comment]
Is this redundant now that https://github.com/oxidecomputer/typify/pull/137 has been merged?

**Comment by ahl** [comment]
Fixed by #137

**Pull request closed without merging**